### PR TITLE
Support application credential when get token

### PR DESCRIPTION
**Pull request by greatbn**

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -161,8 +164,17 @@ func (c *Client) Do(ctx context.Context, req *http.Request) (resp *http.Response
 	}
 
 	// If 401, get new token and retry one time.
+	tcr := &TokenCreateRequest{}
 	if resp.StatusCode == http.StatusUnauthorized {
-		tok, tokErr := c.Token.Create(ctx, &TokenCreateRequest{Username: c.username, Password: c.password})
+		switch c.authMethod {
```
Comment (cuonglm) [inline client.go:169]: Don't do this here, do it in Token.Create instead

File: client.go [hunk 2]
```diff
@@ -190,6 +202,7 @@ func (c *Client) SetKeystoneToken(s string) {
 type ListOptions struct{}
 
 func errorFromStatus(code int, msg string) error {
+	fmt.Print(msg)
```
Comment (cuonglm) [inline client.go:205]: remove this.

File: client.go (hunk 1 above)
Comment (greatbn) [reply client.go:169]: Then we need to pass `c *Client` to `Token.Create` function ?

Comment (cuonglm) [reply client.go:169]: There has been `t.client` already.

File: token_test.go [hunk 3]
```diff
@@ -13,7 +13,7 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-func TestToken(t *testing.T) {
+func TestTokenAuthPassword(t *testing.T) {
```
Comment (cuonglm) [inline token_test.go:16]: Add test for other methods?

File: client.go [hunk 4]
```diff
@@ -162,7 +165,8 @@ func (c *Client) Do(ctx context.Context, req *http.Request) (resp *http.Response
 
 	// If 401, get new token and retry one time.
 	if resp.StatusCode == http.StatusUnauthorized {
-		tok, tokErr := c.Token.Create(ctx, &TokenCreateRequest{Username: c.username, Password: c.password})
+		tcr := &TokenCreateRequest{AuthMethod: c.authMethod, Username: c.username, Password: c.password, AppCredID: c.appCredID, AppCredSecret: c.appCredSecret}
```
Comment (cuonglm) [inline client.go:168]: Then you have to set other information inside `c.Token.Create`

Comment (greatbn) [reply client.go:168]: what information?

File: token_test.go (hunk 3 above)
Comment (greatbn) [reply token_test.go:16]: Yes. For Password and Application Credential.

File: client.go (hunk 4 above)
Comment (cuonglm) [reply client.go:168]: https://github.com/bizflycloud/gobizfly/blob/aa1015fb9f07ee74ad943c18e960ab10b774e012/token.go#L56

Comment (greatbn) [reply client.go:168]: Ah. I see. Thanks

**Pull request merged**